lib/_http_agent.js

```diff
@@ -152,7 +152,15 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     options.servername = options.host;
     const hostHeader = req.getHeader('host');
     if (hostHeader) {
-      options.servername = hostHeader.replace(/:.*$/, '');
+      // abc => abc
+      // abc:123 => abc
+      // [::1] => ::1
+      // [::1]:123 => ::1
+      if (hostHeader.startsWith('[')) {
+        options.servername = hostHeader.split(']')[0].substr(1);
```

Comment:
`.split(']', 1)` is more efficient.

For bonus performance points, if you're going to slice the string anyway, call `.indexOf()` and then `.substr(1, index)` (and default `index` to `hostHeader.length` if it's -1.)

Lastly, I would say that it should pass on the string untouched if it doesn't contain a matching `]`.